application/controllers/ReportController.php

```diff
@@ -192,8 +197,9 @@ public function downloadAction()
 
         switch ($type) {
             case 'pdf':
-                /** @var Hook\PdfexportHook */
-                Pdfexport::first()->streamPdfFromHtml($this->report->toPdf(), $name);
+                /** @var Hook\PdfexportHook $exports */
+                $exports = Pdfexport::first();
+                $exports->streamPdfFromHtml((string) $this->report->toPdf(), $name);
```

Comment:
`PdfExport::streamPdfFromHtml()` expects `string|PrintableHtmlDocument`.

Reply:
?

Reply:
As discussed offline, the abstract class needs to be fixed. But still `PdfExport::streamPdfFromHtml()` also checks if $html is a `PrintableHtmlDocument` instance. In your case it is now always false.